src/aiperf/dataset/graph/workload_detect.py

```
@@ -0,0 +1,530 @@
+    from aiperf.plugin.enums import PluginType
+    from aiperf.plugin.schema.schemas import GraphAdapterMetadata
+
+    matches: list[tuple[int, str]] = []
+    for entry, cls in plugins.iter_all(PluginType.GRAPH_ADAPTER):
+        if entry.name in _AUTODETECT_EXCLUDED:
+            continue
+        # Isolate each adapter's sniff (mirrors ``parser.detect_format``): one
+        # crashing adapter must not abort detection for all the others.
+        try:
+            recognized = cls.can_load(path)
+        except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)